### PR TITLE
Added edge cases, implemented unit test and port ranges

### PR DESCRIPTION
**Pull request by hymeisa**
thank you to secure coding for teaching regular expressions
note: re ships in the base python installation and the dependencies in the docs do NOT need to be updated.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -43,7 +44,8 @@ def verify_ports(ports: str) -> tuple[int, ...]:
 
     Args:
         ports (str):
-            A comma-separated list of port values.
+            A comma-separated list of port values, either single-valued or
+            given as ranges from a lower bound to an upper bound.
```
Comment (sschultz9) [inline src/main.py]: change "or given as ranges" to "or given as a range".

Comment (sschultz9) [inline src/main.py]: Add inclusive to the end of this so that they know the port range is inclusive.

**Review by sschultz9 — CHANGES_REQUESTED** [review]
